src/components/CheckboxInput/CheckboxInput.js

```diff
@@ -0,0 +1,174 @@
+import React, { Fragment } from 'react';
+import PropTypes from 'prop-types';
+import styled, { css } from 'react-emotion';
+import { hideVisually } from 'polished';
+import { shadowSingle, disableVisually } from '../../styles/style-helpers';
+import { childrenPropType } from '../../util/shared-prop-types';
+
+const checkmarkSvg = fill =>
+  // NOTE: Optimizing SVGs in data URIs, https://codepen.io/tigt/post/optimizing-svgs-in-data-uris
+  `data:image/svg+xml,${encodeURIComponent(
+    `<svg width='10' height='10' xmlns='http://www.w3.org/2000/svg'><path d='M3.438 6.973l5.097-5.694a.858.858 0 0 1 1.175-.086c.349.288.389.793.089 1.128l-5.73 6.4a.856.856 0 0 1-1.264 0L.201 5.812a.778.778 0 0 1 .09-1.128.858.858 0 0 1 1.174.086l1.973 2.203z' fill='${fill}' fill-rule='nonzero'/></svg>`
+  )}`;
+
+const baseStyles = ({ theme }) => css`
+  label: Checkbox-input__label;
```

Comment:
Hm, so far, I've stuck to all lowercase BEM classes with the labels. So here I would have chosen `checkbox-input__label`. I think unless we drop hypens, we should stick to using kebab for B, E, and M parts of the class?